packages/at_secondary_server/lib/src/verb/handler/lookup_verb_handler.dart

```diff
@@ -56,105 +56,193 @@ class LookupVerbHandler extends AbstractVerbHandler {
+  ///     is initiated to the [keyOwnersAtSign] to acquire the data.
+  ///
+  ///   - If the user is authenticated using the legacy PKAM, the data is returned.
+  ///   - If the user is authenticated using the APKAM, the function fetches the
+  ///     enrollmentId from metadata. Data is only returned if the atSign has
+  ///     permission to access the key's namespace.
+  ///   - If permissions are not met, [UnAuthorizedException] is thrown.
+  Future<void> _handleAuthenticatedConnection(
+      String keyOwnersAtSign,
+      String thisServersAtSign,
+      String keyAtAtSign,
+      InboundConnection atConnection,
+      Response response,
+      String? operation,
+      AtAccessLog? atAccessLog,
+      String? byPassCacheStr) async {
+    var lookupKey = '$thisServersAtSign:$keyAtAtSign';
+    bool isAuthorized = await _isAuthorizedToViewData(atConnection, lookupKey);
+    if (!isAuthorized) {
+      throw UnAuthorizedException(
+          'Enrollment Id: ${(atConnection.getMetaData() as InboundConnectionMetadata).enrollmentId} is not authorized for lookup operation on the key: $lookupKey');
+    }
+    if (keyOwnersAtSign == thisServersAtSign) {
+      // We're looking up data owned by this server's atSign
+      await _fetchDataOwnedByThisAtSign(thisServersAtSign, keyAtAtSign,
+          atConnection, response, operation, atAccessLog, keyOwnersAtSign);
     } else {
-      // isAuthenticated is false
-      // If the Connection is unauthenticated form the key based on presence of "fromAtSign"
-      var keyPrefix = '';
-      if (!(atConnectionMetadata.isAuthenticated)) {
-        keyPrefix = (fromAtSign == null || fromAtSign == '')
-            ? 'public:'
-            : '$fromAtSign:';
+      // keyOwnersAtSign != thisServersAtSign
+      // We're looking up data owned by another atSign.
+      await _fetchDataOwnedByOtherAtSign(
+          thisServersAtSign, keyAtAtSign, response, operation, byPassCacheStr);
+    }
+  }
+
+  /// Handles requests from unauthenticated or pol-authenticated connections.
+  ///
+  ///   - In the case of an unauthenticated connection,  the "lookup" verb is used
+  ///     to fetch public keys from the storage associated with the current "atSign".
+  ///
+  ///     For instance, if the current atSign is "alice", then using "lookup:phone.wavi@alice"
+  ///     yields the corresponding value of the key "public:phone.wavi@alice".
+  ///
+  ///   - In cases where the 'atSign' does not possess ownership of the data,
+  ///     a pol authentication is initiated towards the 'atSign' that does own the data.
+  ///
+  ///     For instance, when the current "atSign" is "alice" and the request is
+  ///     "lookup:phone.wavi@bob"(from an authenticated connection), an interaction
+  ///     occurs where Alice proves identity to Bob, who subsequently returns the
+  ///     value of the key "@alice:phone.wavi@bob".
+  Future<void> _handleUnAuthAndPolAuthConnection(
```

Comment:
I'd like to see code which is specifically checking the isPolAuthenticated flag. Ideally always there would be three clear cases handled in a verb handler: (1) unauthenticated (2) authenticated (3) polAuthenticated